design/specs/column-extraction.md

```diff
@@ -8,87 +8,58 @@ editor: markdown
 
-3. **The user clicks on the 'Extract Columns' option**
+![image](/assets/design/specs/column-extraction/185140900-ea1514c3-2423-42ad-ab98-a4f87848f463.png)
 
-    The user can choose to keep copies of the columns in the table if needed. By default the extracted columns are deleted.
+### Impact of the Extract Columns Operation
 
-4. **The user inspects the new foreign key column (link)**
+Since the split operation will affect the original table, the user should be made aware of the operation's impact. The dialog will make the following information clear:
 
-    The new foreign key column will contain links to the new records that have been created. Additionally, when selected, the inspector panel will display the link properties.
+- The original table will be updated to remove the extracted columns
+- A new table will be created with the extracted columns
+- A foreign key will be created between the new table and the original table
 
-    <img width="1469" alt="image" src="https://user-images.githubusercontent.com/845767/177546415-23b11ded-8987-4e69-ab1a-1decb9e773d3.png">
+### Link Column Name Convention
 
+To help users understand the role of the link column, the name of the link column will follow a convention that will be applied when users name the new table. The convention will be:
 
-5. **The user navigates to the new table**
+`table_name` `+` `Id`
 
-    The new table will be listed in the list of tables for the current schema. The user can modify the table, add new columns or rename it if needed.
+![image](/assets/design/specs/column-extraction/185142290-db3d6fc7-86b6-4a2d-8c23-7ed48104c8cc.png)
 
-    <img width="874" alt="image" src="https://user-images.githubusercontent.com/845767/177550009-30fcae81-2ae9-4539-ad85-5a0d1289790b.png">
+### Allowing the user to modify the original column selection
 
-# Unresolved Design Problems
+Under the `Columns to Extract` section, the user can extend the original column selection by selecting additional columns. The user can also remove columns from the selection.
 
-## Extracting a foreign key column to create a mapping table
-If a user had a foreign key referencing a publisher in a publications table and wanted to link to more than one publisher, could they select both the table's primary key and that foreign key and create a new table from both? Should there be a special action that does this without adding a foreign key in place of the columns?
+This component could be reused for all actions involving column subsets.
 
-## Suggesting columns to be extracted
-Offering suggestions for columns that can be extracted into new tables would improve the feature's discoverability and provide users with a clear path to a better data model.
+Additionally, the user can rename the columns that will be extracted in cases where the column name is not descriptive enough.
```

Comment:
I'm not sure if the backend supports renaming the column. @silentninja would know. If we don't, then I suggest implementing this renaming feature in the future, we can make a ticket to track the work.

Reply:
We won't be renaming for now. We can create a ticket in the future when needed.